### PR TITLE
Better handeling of submodules

### PR DESCRIPTION
**Pull request by dbeniamine**
Hello,

Your plugin is quite nice, I added a few things to manage properly git
submodules (clean commits and move file).

I also plan to add a remove function and create the vim documentation.

Hoping you'll be interested by my work,
Bests regards,

David

### CODE REVIEW

**Comment by dbeniamine** [comment]
Hum I already added the doc and the Destruct command (It was quicker than what I
thought), still I haven't tested Destruct without git submodule yet.

**Comment by dbeniamine** [comment]
Well it seems that I added a few others features: 

+ Upgrade command (actually I had the code since a while but I was too lazy to
create a full plugin), works also with vim.org plugins, the user just have to
add a two lines files (script url, version number (0 for init))
+ Possibility to look at the Readme in Invoke command
+ Possibility to set the Scry result list length

Hope you'll like it :)

**Comment by ardagnir** [comment]
This looks really awesome!
There's a lot here and I'm really busy right now, but I'll try to look it over soon.

File: plugin/vizardry.vim [hunk 1]
```diff
@@ -384,6 +502,9 @@ endfunction
 
 let s:scriptDir = expand('<sfile>:p:h')
 let s:bundleDir = substitute(s:scriptDir, '/[^/]*/[^/]*$', '', '')
+let s:EvolveVimOrgPath = s:scriptDir.'/EvolveVimOrgPlugins.sh'
+let s:relativeBundleDir=substitute(s:bundleDir,g:VizardryGitBaseDir,'','')
+let s:relativeBundleDir=substitute(s:relativeBundleDir,'^/','','')
```
Comment (ardagnir) [inline plugin/vizardry.vim]: These two previous lines break vizardry if you're not using submodules.

**Comment by dbeniamine** [comment]
Those lines ? 

    let s:scriptDir = expand('<sfile>:p:h')
    let s:bundleDir = substitute(s:scriptDir, '/[^/]*/[^/]*$', '', '')

Aren't they exactly the same in your version ?
Can you tell me more about the bug so I can fix it ?

File: plugin/vizardry.vim (hunk 1 above)
Comment (dbeniamine) [inline plugin/vizardry.vim]: Those lines ?

```
let s:scriptDir = expand('<sfile>:p:h')
let s:bundleDir = substitute(s:scriptDir, '/[^/]*/[^/]*$', '', '')
```

Aren't they exactly the same in your version ?
Can you tell me more about the bug so I can fix it ?

Comment (ardagnir) [inline plugin/vizardry.vim]: No, lines 506 and 507

Error detected while processing /home/me/.vim/bundle/vizardry/plugin/vizardry.vim:
line  506:
E121: Undefined variable: g:VizardryGitBaseDir
E11Error detected while processing /home/me/.vim/bundle/vizardry/plugin/vizardry.vim:
line  506:
E121: Undefined variable: g:VizardryGitBaseDir
E116: Invalid arguments for function substitute(s:bundleDir,g:VizardryGitBaseDir,'','')
E15: Invalid expression: substitute(s:bundleDir,g:VizardryGitBaseDir,'','')
line  507:
E121: Undefined variable: s:relativeBundleDir
E116: Invalid arguments for function substitute(s:relativeBundleDir,'^/','','')
E15: Invalid expression: substitute(s:relativeBundleDir,'^/','','')
Press ENTER or type command to continue6: Invalid arguments for function substitute(s:bundleDir,g:VizardryGitBaseDir,'','')
E15: Invalid expression: substitute(s:bundleDir,g:VizardryGitBaseDir,'','')
line  507:
E121: Undefined variable: s:relativeBundleDir
E116: Invalid arguments for function substitute(s:relativeBundleDir,'^/','','')
E15: Invalid expression: substitute(s:relativeBundleDir,'^/','','')
Press ENTER or type command to continue

Comment (dbeniamine) [inline plugin/vizardry.vim]: Indeed the relativeBundleDir is useless if submodule are not used (Bundle dire relative to git base dir).
The bug is fixed thanks for the report.

**Comment by ardagnir** [comment]
I added a link to your fork in the readme

**Comment by dbeniamine** [comment]
Thanks for the link :) .

**Pull request closed without merging**